client/me/purchases/remove-purchase/index.jsx

```diff
@@ -212,6 +230,43 @@ class RemovePurchase extends Component {
 		);
 	}
 
+	shouldShowPreCancelationDialog() {
+		const { purchase } = this.props;
+		return (
+			config( 'pre_cancelation_modal_enabled' ) &&
+			isPlan( purchase ) &&
+			! isJetpackPlan( purchase ) &&
+			! isGSuiteOrGoogleWorkspace( purchase )
+		);
+	}
+
+	renderPreCancelationDialog() {
+		const { site, purchase, primaryDomain } = this.props;
+		const primaryDomainName = hasCustomDomain && primaryDomain ? primaryDomain.name : '';
+
+		let wordpressComURL = '';
+		if ( typeof site.wpcom_url !== 'undefined' ) {
+			wordpressComURL =
+				site.wpcom_url.length < 35
+					? site.wpcom_url
+					: site.wpcom_url.substr( 0, 10 ) + '…' + site.wpcom_url.slice( -20 );
+		}
```

Comment:
I removed that logic from the parent components, at the moment I didn't ported it to the dialog, keeping it lighter showing the static placeholder text:  Warning: domain traffic will be redirected

The proper text will be added back with the copy PR.

Also I don't think we can solve this with `CSS overflow`  as
```
overflow: hidden;
text-overflow: ellipsis;
```
Because they requested ellipsis in the middle of the domain name, not at its end.